### PR TITLE
Revamp the landing page with cards

### PR DESCRIPTION
**Pull request by wtbarnes**
This is a redesign of the landing page with [`sphinx-design`](https://sphinx-design.readthedocs.io). Thoughts on the overall look are welcome.

Fixes https://github.com/sunpy/sunpy/issues/6641

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -2,63 +2,97 @@
...
 
-.. todo::
+    .. grid-item-card:: Known Issues
+        :link: known_issues
+        :link-type: ref
+        :text-align: center
 
-    When a page or subgallery containing more diataxis like how-to (longer form) guides is added then link here
-    [How to guides] provide detailed overview on some of the features of sunpy. These assume some level of experience using sunpy.
+        :material-outlined:`wb_sunny;8em`
```
Comment (ayshih) [inline docs/index.rst:81]:
```suggestion
        :material-outlined:`sentiment_dissatisfied;8em`
```
Heh...

File: docs/index.rst [hunk 2]
```diff
@@ -2,63 +2,97 @@
...
-    Index explanation pages (e.g. different map rotation methods and then link here.)
-    [Topic guides] discuss key topics and concepts at a fairly high level and provide useful background information and explanation.
+    .. grid-item-card:: Developer's Guide
+        :link: dev_guide
+        :link-type: ref
+        :text-align: center
 
-Further Info
-============
+        :material-outlined:`wb_sunny;8em`
```
Comment (ayshih) [inline docs/index.rst:90]:
```suggestion
        :material-outlined:`construction;8em`
```

File: docs/index.rst [hunk 3]
```diff
@@ -2,63 +2,97 @@
...
 
-.. todo::
+    .. grid-item-card:: Get Help
+        :link: whatsnew
+        :link-type: ref
+        :text-align: center
 
-    if #6615 is completed link here
-    Introduction to sunpy core
+        :material-outlined:`support;8em`
```
Comment (ayshih) [inline docs/index.rst:63]:
```suggestion
        :material-outlined:`live_help;8em`
```

**Comment by nabobalis** [comment]
I assume the how-to part was not meant to be here?

**Comment by Cadair** [comment]
The get help button text is too small I think.

File: docs/index.rst [hunk 4]
```diff
@@ -2,63 +2,89 @@
 sunpy core Documentation
 ************************
 
+.. image:: tour.svg
+  :align: center
+
 ``sunpy`` is a community-developed, free and open-source solar data analysis environment for Python.
 It provides search and download functionality, data containers for image and time series data, as well as commonly used coordinate frames and transformations between such frames.
```
Comment (Cadair) [inline docs/index.rst:9]: I feel like this sentence is missing a "and a bunch of other stuff" line lol

Comment (wtbarnes) [reply docs/index.rst:9]: I think you commented on the wrong line and I'm not quite sure what you mean?

File: docs/index.rst [hunk 5]
```diff
@@ -2,63 +2,89 @@
...
-* View a list of :ref:`known issues <known_issues>` or report new issues on our `GitHub issue tracker`_.
-* Join our `chat room`_ or post to our `discourse`_ for help.
-* Read the :ref:`Developers guide <dev_guide>` to start contributing to the SunPy project.
 
+.. button-link:: https://community.openastronomy.org/c/sunpy/5
 
-.. _chat room: https://app.element.io/#/room/#sunpy:openastronomy.org
-.. _GitHub issue tracker: https://github.com/sunpy/sunpy/issues
-.. _discourse: https://community.openastronomy.org/c/sunpy/5
+    Post on Discourse
```
[the remark this answers is not in the document]
Comment (wtbarnes) [reply docs/index.rst:87]: Ha oops. Do not know how that was left.

File: docs/conf.py [hunk 6]
```diff
@@ -119,6 +119,7 @@
     'sphinx_design',
     'sphinx_copybutton',
     'hoverxref.extension',
+    'sphinx_design',
```
Comment (nabobalis) [inline docs/conf.py:122]:
```suggestion
```

File: docs/conf.py [hunk 7]
```diff
@@ -239,6 +240,8 @@
     "term": "tooltip",
 }
 
+# Panels options
+
```
Comment (nabobalis) [inline docs/conf.py:244]: Do we need this?

File: docs/index.rst [hunk 8]
```diff
@@ -2,54 +2,84 @@
 sunpy core Documentation
 ************************
 
+.. image:: tour.svg
+  :align: center
+
 ``sunpy`` is a community-developed, free and open-source solar data analysis environment for Python.
 It provides search and download functionality, data containers for image and time series data, as well as commonly used coordinate frames and transformations between such frames.
```
Comment (nabobalis) [inline docs/index.rst:9]:
```suggestion
``sunpy`` among other things provides search and download functionality, data containers for image and time series data, as well as commonly used coordinate frames and transformations between such frames.
```
I think this is what Stuart wanted. Tho my corrections are bad. The sentence suggests we only do those three things.

File: docs/index.rst [hunk 9]
```diff
@@ -2,54 +2,84 @@
...
+        **New users start here!** Walkthrough on how to install sunpy and use the key features of the package.
+
+    .. grid-item-card:: How-to Guides
+        :link: how_to_guide
+        :link-type: ref
+        :text-align: center
+
+        :material-outlined:`menu_book;8em;sd-text-secondary`
+
+        Snippets of code for accomplishing specific tasks with sunpy. Most useful for answering "How do I..." questions.
```
Comment (nabobalis) [inline docs/index.rst:30]:
```suggestion
        Snippets of code for accomplishing specific tasks with sunpy.
        Most useful for answering "How do I..." questions.
```
Hopefully this won't break.

File: docs/index.rst [hunk 10]
```diff
@@ -2,54 +2,84 @@
...
+    .. grid-item-card:: Topic Guides
+        :link: guide
+        :link-type: ref
+        :text-align: center
+
+        :material-outlined:`school;8em;sd-text-secondary`
 
-New Users
-==========
+        In-depth explanations of concepts and key topics. Most useful for answering "why" questions.
```
Comment (nabobalis) [inline docs/index.rst:47]:
```suggestion
        In-depth explanations of concepts and key topics.
        Most useful for answering "why" questions.
```

File: docs/index.rst (hunk 8 above)
Comment (wtbarnes) [reply docs/index.rst:9]: How about just changing "provides" to "includes"? That perhaps is a bit less exclusionary. Also, is there any major feature of the package we are missing with this description?

File: docs/conf.py (hunk 7 above)
Comment (wtbarnes) [reply docs/conf.py:243]: No, this was left over from my panels experiment.

**Comment by wtbarnes** [comment]
It is annoying that you have to scroll down to see the panels. I wonder if it would be better to delete the infographic. I don't think it serves much purpose...

**Comment by nabobalis** [comment]
> It is annoying that you have to scroll down to see the panels. I wonder if it would be better to delete the infographic. I don't think it serves much purpose...

Maybe move it under the boxes or maybe to the intro page?

File: docs/index.rst (hunk 8 above)
Comment (wtbarnes) [reply docs/index.rst]: I took another stab at it. Let me know what you think (and maybe this makes @Cadair a bit happier too)

**Review by nabobalis — APPROVED** [review]

**Review by dstansby — APPROVED** [review]

**Pull request merged**